librz/search/collection.c

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: 2024 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2024 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_search.h>
+#include "search_internal.h"
+
+/**
+ * \brief      Initialize a new RzSearchCollection
+ *
+ * \param[in]  search_over  The search over callback to set
+ * \param[in]  free         The free function to set
+ *
+ * \return     On success returns a valid pointer, otherwise NULL.
+ */
+RZ_IPI RZ_OWN RzSearchCollection *rz_search_collection_new(RZ_NONNULL RzSearchOverCallback search_over, RZ_NULLABLE RzPVectorFree free) {
```

Comment:
```suggestion
RZ_IPI RZ_OWN RzSearchCollection *rz_search_collection_new(RZ_NONNULL RzSearchOverCallback search_over, RZ_NULLABLE RzPVectorFree element_free) {
```

Or something similar.